analytics/aggregate.go

```diff
@@ -20,6 +20,9 @@ const (
 	AggregateSQLTable             = "tyk_aggregated"
 )
 
+// lastDocumentTimestamp is a map to store the last document timestamps of different Mongo Aggregators
+var lastDocumentTimestamp = make(map[string]time.Time)
```

Comment:
@mativm02 why is this a global variable? What would happen if we have multiple mongo-aggregate and sql-aggregate pumps running at the same time?

Reply:
@tbuchaillot because the function `AggregateData` is being used by Mongo and SQL. I could add a `lastDocumentTimestamp` field to `MongoAggregateConf` for example, and send it to to `AggregateData` through a parameter in order to get the last timestamp of Mongo instance.
The problem I noticed is that SQL wouldn't have this field in their config struct. That's why I've created this variable as global. We may think a better way to store this data.

The idea of the map is to organise every Mongo instance by their MongoURL (that's the key of the map), and the value is the last aggregation timestamp. So if we have multiple Mongo instances running, all of them we'll be stored in the map by their MongoURL. 
This feature is not being applied to SQL yet.

Reply:
@mativm02 my point here is that you can have multiple pumps running and aggregating data at the same time.  Using a global var with concurrent writes can have some problems so I'd suggest switching to a sync.Map or use mutexes here.

Reply:
Also, for SQL cases maybe it's worth using the connection string? Imagine that you have 2 SQL aggregate pumps writing in different dbs.